modules/table/src/integrationTest/java/org/apache/ignite/internal/disaster/ItDisasterRecoveryReconfigurationTest.java

```diff
@@ -1234,6 +1242,295 @@ void testTwoPhaseResetOnEmptyNodes() throws Exception {
+                Assignment.forPeer(node(0).name()),
+                Assignment.forPeer(node(1).name()),
+                Assignment.forPeer(node(2).name()),
+                Assignment.forPeer(node(3).name()),
+                Assignment.forPeer(node(4).name()),
+                Assignment.forPeer(node(5).name()),
+                Assignment.forPeer(node(6).name())
+        ), timestamp);
+
+        assertStableAssignments(node0, partId, allAssignments);
+
+        // Assignments chain is equal to the stable assignments.
+        assertAssignmentsChain(node0, partId, AssignmentsChain.of(allAssignments));
+
+        // Write data(1) to all nodes.
+        List<Throwable> errors = insertValues(table, partId, 0);
+        assertThat(errors, is(empty()));
+
+        Assignments link2Assignments = Assignments.of(Set.of(
+                Assignment.forPeer(node(0).name()),
+                Assignment.forPeer(node(1).name()),
+                Assignment.forPeer(node(2).name())
+        ), timestamp);
+
+        AtomicBoolean blockedLink2 = new AtomicBoolean(true);
+
+        // Block stable switch to check that we initially add reset phase 1 assignments to the chain.
+        blockMessage((nodeName, msg) -> blockedLink2.get() && stableKeySwitchMessage(msg, partId, link2Assignments));
+
+        logger().info("Stopping nodes [ids={}].", Arrays.toString(new int[]{3, 4, 5, 6}));
+
+        stopNodesInParallel(3, 4, 5, 6);
+
+        Assignments link2FirstPhaseReset = Assignments.of(Set.of(
+                Assignment.forPeer(node(0).name())
+        ), timestamp);
+
+        assertStableAssignments(node0, partId, link2FirstPhaseReset, 60_000);
+
+        // Assignments chain consists of stable and the first phase of reset.
+        assertAssignmentsChain(node0, partId, AssignmentsChain.of(List.of(allAssignments, link2FirstPhaseReset)));
+
+        // Unblock stable switch, wait for reset phase 2 assignments to replace phase 1 assignments in the chain.
+        blockedLink2.set(false);
+
+        assertStableAssignments(node0, partId, link2Assignments, 30_000);
+
+        // Assignments chain consists of stable and the second phase of reset.
+        assertAssignmentsChain(node0, partId, AssignmentsChain.of(List.of(allAssignments, link2Assignments)));
+
+        logger().info("Stopping nodes [ids={}].", Arrays.toString(new int[]{1, 2}));
+        stopNodesInParallel(1, 2);
+
+        Assignments link3Assignments = Assignments.of(Set.of(
+                Assignment.forPeer(node(0).name())
+        ), timestamp);
+
+        assertStableAssignments(node0, partId, link3Assignments, 30_000);
+
+        assertAssignmentsChain(node0, partId, AssignmentsChain.of(List.of(allAssignments, link2Assignments, link3Assignments)));
+    }
+
+    @Test
+    @ZoneParams(nodes = 7, replicas = 7, partitions = 1, consistencyMode = ConsistencyMode.HIGH_AVAILABILITY)
+    void testSecondResetRewritesFirst() throws Exception {
```

Comment:
The name is misleading. second phaze of reset should rewrite first one, on the other hand second reset(full) should append to the first one.

Reply:
Yep, good.